services/core-api/app/api/services/email_service.py

```diff
@@ -0,0 +1,176 @@
+import requests, json
+
+from enum import Enum
+from flask import current_app
+
+from app.config import Config
+from app.api.constants import CORE_PURPLE_LOGO_BASE64_ENCODED
+
+
+class EmailBodyType(Enum):
+    HTML = 'html'
+    TEXT = 'text'
+
+
+class EmailEncoding(Enum):
+    BASE64 = 'base64'
+    BINARY = 'binary'
+    HEX = 'hex'
+    UTF8 = 'utf-8'
+
+
+class EmailPriority(Enum):
+    LOW = 'low'
+    NORMAL = 'normal'
+    HIGH = 'high'
+
+
+MDS_NO_REPLY_SIGNATURE = f'''
+<div>
+    <hr />
+    <p>This is a no-reply email address. If you need to contact the MDS team, please email us at: <a href="mailto: {Config.MDS_EMAIL}">{Config.MDS_EMAIL}</a>.</p>
+    <br />
+    <img src="{CORE_PURPLE_LOGO_BASE64_ENCODED}" width="320" height="106" alt="Core Logo">
+</div>
+'''
+
+
+# NOTE: Manage our Common Services application access here: https://getok.nrs.gov.bc.ca/app/apps/MDS
+# NOTE: See here for details: https://ches.nrs.gov.bc.ca/api/v1/docs
+class EmailService():
+    '''Service wrapper for the Common Services "Common Hosted Email Service" (CHES) email service.'''
+
+    # NOTE: See here for details: https://ches.nrs.gov.bc.ca/api/v1/docs#section/Authentication
+    @classmethod
+    def get_auth_token(cls):
+        '''Gets an auth token required for all requests.'''
+
+        url = Config.COMMON_SERVICES_AUTH_HOST
+        data = {'grant_type': 'client_credentials'}
+        auth = (Config.COMMON_SERVICES_CLIENT_ID, Config.COMMON_SERVICES_CLIENT_SECRET)
+        resp = requests.post(url, data, auth=auth)
+        try:
+            resp_data = resp.json()
+        except ValueError:
+            resp_data = None
+
+        if resp.status_code != requests.codes.ok:
+            message = f'Common Services authentication request returned {resp.status_code}.'
+            if resp_data:
+                message += f'\nError: {resp_data.get("error")}\nDescription: {resp_data.get("error_description")}'
+                current_app.logger.debug(resp_data)
+            current_app.logger.error(message)
+            return
+
+        auth_token = resp_data.get('access_token')
+        if not auth_token:
+            message = 'Common Services authentication request did not return an access token!'
+            current_app.logger.debug(resp_data)
+            current_app.logger.error(message)
+            return
+
+        return auth_token
+
+    # NOTE: See here for details: https://ches.nrs.gov.bc.ca/api/v1/docs#tag/Health
+    @classmethod
+    def perform_health_check(cls):
```

Comment:
Should not return failure or success the health check function called in send_email function?

Reply:
We still want to attempt to send the email even if a dependency in the health check is not healthy (in my opinion). So there is no use at the moment for this to return anything. We just call it before we send the email and then it logs any issues.
For example, maybe the SMTP server dependency is unhealthy, but their queue service is healthy. We still want to send the email request and it will probably get sent out later.